src/data-workspace/validation/use-validation-result.js

```diff
@@ -0,0 +1,118 @@
+import { useCallback } from 'react'
+import { useIsMutating, useQueries } from 'react-query'
+import {
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../../context-selection/index.js'
+import { useApiAttributeParams } from '../../shared/index.js'
+
+const getValidationMetaDataQueryKey = (datasetId) => {
+    const queryKey = [
+        `validationRules`,
+        {
+            params: {
+                dataSet: datasetId,
+                fields: [
+                    'id',
+                    'importance',
+                    'operator',
+                    'leftSide',
+                    'rightSide',
+                    'displayInstruction',
+                    'displayDescription',
+                    'displayName',
+                ],
+            },
+        },
+    ]
+
+    return queryKey
+}
+
+const getValidationQueryKey = (
+    datasetId,
+    { periodId, orgUnitId, categoryComboId, categoryOptionIds }
+) => {
+    const params = {
+        pe: periodId,
+        ou: orgUnitId,
+        cc: categoryComboId,
+        cp: categoryOptionIds?.join(';'),
```

Comment:
Have you verified if these two parameters work with this API? I haven't seen this in the API docs and so far I haven't found a dataset with a non-default attribute combo _and_ val'n rule violations to test it out, but what I would look for is that this filters the validation rule violations in the response by the provided attribute combo and options

Maybe I can set up a dataset to test this out

Reply:
Sure! I tried it out on the "Mortality < 5 years" data set on debug/dev - it takes opening up the data set's options page in the Maintenance app ([link](https://debug.dhis2.org/dev/dhis-web-maintenance/index.html#/edit/dataSetSection/dataSet/pBOMPrpg1QX)), and I set the "Category combination" option to "Target vs. Result"

Then I opened up that data set in the DE app ([link on localhost](http://localhost:3000/#/?attributeOptionComboSelection=eLwL77Z9E7R-STgy27fGvaU&dataSetId=pBOMPrpg1QX&orgUnitId=DiszpKrYNg8&periodId=202202)) and ran validation:
- I noticed that the `api/validation` response returns rule violations for the `default` attribute option combo, and not for the one currently selected (i.e., not for the "Target" selection in "Target vs. Result" in the link above)
- Editing the meningitis value for the "Target" attribute option combo to create a rule violation doesn't work (or at least it isn't reflected in the `validation` response

So I think this means that this request isn't correctly filtered by attribute option combo, and we either need to either 1) figure out how to _actually_ do that with this endpoint, 2) set up a new endpoint where it's possible, and make a note  (Jira issue?) to come back and implement this once that endpoint is ready

I realized that you might not have gotten a chance to learn what things like "attribute option combo" mean since you've been here, which can be a little complex -- in case you haven't, here's an attempt at a definition: "Attribute option combo" means the "unique ID representing the combination of selected options in each of the `category` parameters (which is configured by the 'category combination' that we modified in the maintenance app) that define the context of this data set". You will see "attribute combo" and "category combo" and other places where either "attribute" or "category" is used -- "attribute" means "category defined at the _data set_ level", to distinguish from the category combinations that are used for individual data elements. Maybe that's helpful? 😅

Reply:
thanks for the explanation - I will ask for a one-to-one walkthrough as it still confuses me though! :) ... I created a Jira ticket for investigating the attribute option combo and see whether we're doing something wrong or need a new endpoint: https://dhis2.atlassian.net/browse/TECH-1287

Reply:
Perfect, thanks for the issue report 🙂